bin/update_dependencies.py

```diff
@@ -12,21 +12,39 @@
 
 # CUSTOM_COMPILE_COMMAND is a pip-compile option that tells users how to
 # regenerate the constraints files
-os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_constraints.py"
+os.environ['CUSTOM_COMPILE_COMMAND'] = "bin/update_dependencies.py"
 subprocess.check_call([
     'pip-compile',
     '--allow-unsafe',
     '--upgrade',
     'cibuildwheel/resources/constraints.in',
 ])
-for python_version in ['27', '35', '36', '37']:
-    subprocess.check_call([
-        f'./env{python_version}/bin/pip-compile',
-        '--allow-unsafe',
-        '--upgrade',
-        'cibuildwheel/resources/constraints.in',
-        '--output-file', f'cibuildwheel/resources/constraints-python{python_version}.txt'
-    ])
+
+if 'GITHUB_ACTIONS' in os.environ:
```

Comment:
Honestly, a docker based solution for local runs wouldn't be so bad either...

Reply:
I'm +1 on this. I didn't want to change the current behavior at first.

Reply:
How about a `--docker` flag instead?

Reply:
Though, if @joerick is okay with it, for simplicity, I'd be happy to have a docker method as the only method.

Reply:
Using a manylinux image is a great idea. But it would be nice if there was still a way to run using local pythons, as I'm about to get an Apple Silicon machine and I'm not sure if i'll be able to run Docker, at least in the short term. `'--local-pythons' in sys.argv` would do the trick

Reply:
@joerick, updated with your proposed change.

PS: docker is available in preview https://docs.docker.com/docker-for-mac/apple-m1/. It should be able to run x86_64 images through emulation.